apis/kueue/webhooks/resourceflavor_webhook.go

```diff
@@ -14,59 +14,72 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package webhooks
 
 import (
+	"context"
+
 	"k8s.io/apimachinery/pkg/apis/meta/v1/validation"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
 )
 
 // log is for logging in this package.
 var resourceFlavorLog = ctrl.Log.WithName("resource-flavor-webhook")
 
-func (rf *ResourceFlavor) SetupWebhookWithManager(mgr ctrl.Manager) error {
+type ResourceFlavorWebhook struct{}
+
+func setupWebhookForResourceFlavor(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(rf).
+		For(&kueue.ResourceFlavor{}).
+		WithDefaulter(&ResourceFlavorWebhook{}).
+		WithValidator(&ResourceFlavorWebhook{}).
 		Complete()
 }
 
 // +kubebuilder:webhook:path=/mutate-kueue-x-k8s-io-v1alpha1-resourceflavor,mutating=true,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=resourceflavors,verbs=create,versions=v1alpha1,name=mresourceflavor.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Defaulter = &ResourceFlavor{}
+var _ webhook.CustomDefaulter = &ResourceFlavorWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *ResourceFlavorWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	rf := obj.(*kueue.ResourceFlavor)
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type
-func (rf *ResourceFlavor) Default() {
 	resourceFlavorLog.Info("defaulter", "resourceFlavor", klog.KObj(rf))
```

Comment:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#remove-string-formatting-from-log-message